# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

My provisional lane is **Refresh / Content Opportunity Scoring**. I chose this lane because I want to identify which pages should be reviewed first based on visibility, decline, freshness, and engagement signals. The final goal is to create a ranked list that helps a content team decide where to focus its limited time.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

The project will help a content or SEO team decide which pages should be reviewed first. The output will be a ranked list of pages with suggested actions such as refresh, improve, protect, prune, or monitor. A wrong recommendation may waste the team's time on a page that does not need attention, while missing an important page may lead to a valuable opportunity being overlooked.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [2]:
import pandas as pd
df = pd.read_csv("https://raw.githubusercontent.com/alihuseynliofficial/flyrank-ml-internship/main/data/raw/content_refresh_anonymized.csv")
declining = df[df["trend_direction"] == "down"]
print("Total pages:", len(df))
print("Declining pages:", len(declining))
print("Declining percentage:", round(len(declining) / len(df) * 100, 2), "%")

Total pages: 30000
Declining pages: 16262
Declining percentage: 54.21 %


The starter dataset contains 30,000 pages. Among them, 16,262 pages, or 54.21%, are marked as declining. This shows that more than half of the pages may require review, so a ranked priority system could help the content team decide which pages to examine first.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This project can identify observed patterns and provide a directional, decision-support ranking of pages that may be worth reviewing first.

It cannot prove that refreshing a page will cause its traffic or ranking to improve. It also cannot predict or explain Google's algorithm. A high score only means that a page deserves human review, not that an update will definitely improve its performance.

## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.